### PR TITLE
Add Drain Cleaner installation files to the operators repo

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Task

### Description

As described in the [SP-29](https://github.com/strimzi/proposals/blob/main/029-adopt-the-drain-cleaner-utility.md), the installation files for the Drain Cleaner project should be added to the operators installation folders. It also adds a README.md file linking it to the Drain Cleaner repository / README.md. In the future, this should probably link to the Strimzi docs.

This PR adds it only to the `packaging/install` folder => it will propagate to the `install/` folder in the next release.

### CODE REVIEW

File: packaging/install/drain-cleaner/kubernetes/040-Secret.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
...S2UyCndEWWxzazFadDdYamJwSkpqVlJ3VjgyaUFRVXM5NWxyd0xkS3c3U1d2RHVXcnFVamMzck0wYlRodVhsRDBRcUcKVGwvczg2TFJBWGtqNFh0eFVaL2ZIS2R4bnNIU1JTa3JhVExSVWJ5ZDErMUxHbHJqMVkwaVJxbkhRVU1Wa1E1Nwo1Y3VTTks1NEN1VWQveVg4RGJpVzI4Ri9CQk1DZ1lFQTA0b0d1dTNXSFovdExyYm8yL3NSMTlrOUF0ZmpNbk1GCmtrSTMxcHBPR204c1NMdmZGWkRJTTRqSlcxMU9NZUNGS1lzOW52ZE9laCtlSnhENGowWTNtSExQbURFRVRuaWYKZzl0TlVwcXp1dGtUcTdPR002VE43Q2toR2lOQjN5SU1SbmVMWUVGVG5VM2Y0OXg1RS9aZDZ5Rm93b3gwVkNIZworcklvb3E0NkFOY0NnWUJYSFh3MHlwdEYwQnoyb0ltL1JPcnR0SGU2Rno1bVdZYXJWV3pFM1Blb1RhSzJIY0FtCnpKYk9CWnJqd1BGY0NzV1ZnMGlEMjNTMlJNWlVZMFF3dFBGaWZjOEpOUjdDcTI1cXFMNjFkMmVSTVBoM0Q4d3kKWTcyekJOU3NOQlhTeTRWMXpNWjBpZXJhM2F3bTNVS1U1RWpqZXVqUTdrSkp5WDBBOURMOW9HNGFid0tCZ0JtNApEbGxyMFR5STFTMTR6NWJ2VDV3T3JYeExBelIvQjQ1Z3ZxazQyWVlkUGlVWnNtWE0xSXB2ZHd2RXJGMGtEd3llCkFUL3lrWGdIb0JMZjNWaU9FUFJZQVNWZjBFWTZKQ3JEM3dMTURFNExWbmpwdXNkTW9qeTBRM0lLRjFncHE1UmwKdEd4ZmhPcjdMTjFuVU5FOGd1NnpnU2ttS242UUgrcGEvdFhJbVFUQkFvR0FPcCt0Rnl5ZG16RFJVZWwxMlhEUApFaFB3bmxDdzNSVjhrUUpjRk00eXhCbC9wQmp5K29vNU1NSlZyOFpQKzc1d1FSWkU5d21PZjk2MUVtV2N0VFZMClNVaDl4R3ljN2diY01JRlVQdXlnenVJNkNjWVhsRCtiR1VpUngrb29hRkNKL0pGVkh6bnRHc0RZY0J4TW13OUcKRFY0U2ZtZnlMeTRyRlFYdSsyRGVvY0k9Ci0tLS0tRU5EIFBSSVZBVEUgS0VZLS0tLS0K
```
Comment (tombentley) [inline packaging/install/drain-cleaner/kubernetes/040-Secret.yaml:11]: Are these hard coded cert+key?

File: packaging/install/drain-cleaner/kubernetes/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Installing Strimzi Drain Cleaner on Kubernetes without CertManager
+
+Strimzi Drain Cleaner is using `ValidationWebhook` to receive information about Strimzi pods being evicted.
```
Comment (tombentley) [inline packaging/install/drain-cleaner/kubernetes/README.md:3]:
```suggestion
Strimzi Drain Cleaner uses a `ValidationWebhook` to receive information about Strimzi pods being evicted.
```

File: packaging/install/drain-cleaner/kubernetes/README.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+Kubernetes requires that `ValidationWebhooks` are secured using TLS.
+So the web-hook service need to have HTTPS support.
```
Comment (tombentley) [inline packaging/install/drain-cleaner/kubernetes/README.md:5]:
```suggestion
So the web-hook service needs to have HTTPS support.
```

File: packaging/install/drain-cleaner/kubernetes/webhook-certificates/build.sh [hunk 4]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+
```
Comment (tombentley) [inline packaging/install/drain-cleaner/kubernetes/webhook-certificates/build.sh:2]: Should we check for the existence of `cfssl` and exit with a message if it's missing?

File: packaging/install/drain-cleaner/kubernetes/webhook-certificates/build.sh [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+# Sign webhook certificate
+cfssl genkey webhook.json | cfssljson -bare webhook
+cfssl sign -config config.json -profile server -ca ca.pem -ca-key ca-key.pem webhook.csr webhook.json | cfssljson -bare webhook
+
+# Create CRT bundles
+cat webhook.pem > webhook-bundle.crt
+cat ca.pem >> webhook-bundle.crt
+
+# Convert keys to PKCS8
+openssl pkcs8 -topk8 -nocrypt -in ca-key.pem -out ca.key
```
Comment (tombentley) [inline packaging/install/drain-cleaner/kubernetes/webhook-certificates/build.sh:15]: Do we have to depend on _both_ cfssl and openssl?

**Review by tombentley — APPROVED** [review]
A few comments, but basically LGTM.

File: packaging/install/drain-cleaner/kubernetes/040-Secret.yaml (hunk 1 above)
Comment (scholzj) [reply packaging/install/drain-cleaner/kubernetes/040-Secret.yaml:11]: Yes, this is the prepared key for the pre-defined namespace. If you would not have it there, everyone would need to first generate the key and update all the resources.

File: packaging/install/drain-cleaner/kubernetes/webhook-certificates/build.sh (hunk 5 above)
Comment (scholzj) [reply packaging/install/drain-cleaner/kubernetes/webhook-certificates/build.sh:15]: It is just an example script ... at the end everyone can generate the certificates anyway they want. They do not have to use these tools.

**Comment by scholzj** [comment]
@tombentley I fixed the README.md and added the checks for the installed programs. I will also open a PR in the Drain Cleaner repo.

I understand that the content of the `kubernetes` folder is not optimal, but I think that having the example including the scripts and certificates is the best possible way to show users how to install it (for those who are not on OCP or do not have Cert Manager installed). If you have some better ideas, we should discuss them.

The script as it is here is as it was released - so I think if we wanna do some further changes to it, we should do it in the Drain Cleaner repo for the next Drain Cleaner release.

WDYT?

**Pull request merged**